### ISSUE TITLE
"Barrier is starting."  But not working.

### ISSUE THREAD

**Issue by gregrwm** [input-leap/input-leap#1514]
### What happened?

Barrier is stuck at "Barrier is starting."  What might i need?

Both boxen have a wire to my modem, are fully updated, and can ssh to each other.  Both have this /etc/barrier.conf:
```
section: screens
        e540:
        eb840g3:
end
section: links
        e540:
                up    = eb840g3
        eb840g3:
                down  = e540
end
```

i invoked barrier(2.4.0-1)(installed by pacman) on e540(manjaro kde), in the terminal it said:
>*** WARNING *** The program 'barrier' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
>*** WARNING *** Please fix your application to use the native API of Avahi!
>*** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>

i found https://github.com/input-leap/input-leap/issues/780 via google but it's not clear to me if the comments solved the issue.  Perhaps if i give the server address to the client it doesn't really need avahi?

Not knowing what to make of that i went ahead in the barrier GUI, designated it as 'Server', selected 'Use existing configuration:' at /etc/barrier.conf and clicked 'Start'.  in the terminal it said:
>("-f", "--no-tray", "--debug", "INFO", "--name", "e540", "--disable-client-cert-checking", "-c", "/etc/barrier.conf", "--address", ":24800")

and in the GUI it says "Barrier is running."  Then i invoked barrier(2.4.0-1.7)(installed by zypper) on 3b840g3(tumbleweed kde), designated it as 'Client', 'Auto config', clicked 'Start', in the terminal it said:
>("-f", "--no-tray", "--debug", "INFO", "--name", "eb840g3", "[]:24800")

and in the GUI it says "Barrier is starting."  Tried uncheck 'Auto config', entered Server IP, clicked 'Reload', 'Stop', and 'Start', in the terminal it said:
>("-f", "--no-tray", "--debug", "INFO", "--name", "eb840g3", "[10.3.8.21]:24800")

and in the GUI it still says "Barrier is starting."

Both GUIs say "SSL Fingerprint: Disabled".  The manjaro default install has no firewall iiuc.  Tumbleweed does, but as client does it need anything open?  i tried "firewall-cmd --add-port=24800/tcp" on tumbleweed, and clicked 'Stop' and 'Start' on the Server and then the Client, but no change.

What else might i check, or read, or do?
Many thanks,

### Version

v2.4.0

### If applicable, where did you install Barrier from?

pacman and zypper, from the default repositories

### What OSes are you seeing the problem on? (Check all that apply)

Linux

### What OS versions are you using?

server on manjaro kde, client on tumbleweed kde, both are rolling release and up to date

### Relevant log output

```shell
manjaro/server:
Sep 20 23:51:31 e540 barrier[15783]: *** WARNING *** The program 'barrier' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
Sep 20 23:51:31 e540 barrier[15783]: *** WARNING *** Please fix your application to use the native API of Avahi!
Sep 20 23:51:31 e540 barrier[15783]: *** WARNING *** For more information see <http://0pointer.de/blog/projects/avahi-compat.html>
Sep 20 23:51:31 e540 dbus-daemon[399]: [system] Activating via systemd: service name='org.freedesktop.Avahi' unit='dbus-org.freedesktop.Avahi.service' requested by ':1.322' (uid=1000 pid=15783 comm="barrier")
Sep 20 23:51:31 e540 dbus-daemon[399]: [system] Activation via systemd failed for unit 'dbus-org.freedesktop.Avahi.service': Unit dbus-org.freedesktop.Avahi.service not found.
Sep 20 23:51:31 e540 kded5[720]: Registering ":1.71/StatusNotifierItem" to system tray

tumbleweed/client:
Sep 20 23:53:04 eb840g3 kded5[1999]: Registering ":1.75/StatusNotifierItem" to system tray
```

**Comment by shymega** [comment]
Whilst Input Leap is forked from Barrier, they may have implemented changes we don't have in this fork. Therefore, we cannot support isues native to Barrier - only Input Leap. Closing as invalid, as we're not the Barrier project. Please open the issue on their tracker. Thank you.

**Comment by gregrwm** [comment]
i'll be delighted to run inputleap instead as soon as it arrives in the manjaro/tumbleweed repos.

**Comment by shymega** [comment]
Yes, and we're still working on the release. When we can. For now, this issue is best filed in the Barrier repo. I know it's looking unmaintained, but it does belong there equally.

**Comment by gregrwm** [comment]
thank you, see https://github.com/debauchee/barrier/issues/1784

**Comment by Reicoler** [comment]
I also have this issue where its just InputLeap is starting. forever. and dosen't actually start

**Comment by shymega** [comment]
OK, reopening.

**Comment by KoreanPizzaGuy** [comment]
MacOs - 13.4.1
Set on server mode - works fine
Set on client mode - stuck on 'InputLeap is starting'

**Comment by kiradotlgbt** [comment]
same here, macos works fine as a server but never starts as a client :(

**Comment by shymega** [comment]
Hi,

Please avoid bumping the issue when unnecessary.

If you also experience the issue, kindly react to the issue with `:+1:`, or if you want to find out when the issue will be resolved, please do not bump the issue, but _instead_ subscribe to the issue using the button on the right-hand side.

**Comment by p12tic** [comment]
@nosbig Thanks for bringing this up. This makes it much clearer what the root cause may be and what configurations would be affected.

**Comment by Pecacheu** [comment]
@nosbig Using Win 11 here (both client and server), and I did a similar process to get it working. I had to force-restart the InputLeap service after setting the IP. (Also had to force-quit any running background processes before reopening.) Then it finally got past the "InputLeap is starting" stage.
